### ISSUE TITLE
Native VS-like Dark Theme Support [Live screenshot included]

### ISSUE THREAD

**Issue by NazmusLabs** [Microsoft/winfile#63]
![windows 3 1 file manger true color on dosbox ss1804](https://user-images.githubusercontent.com/5572377/38568116-dbe6d2f2-3cb5-11e8-8f05-bffb61ab8021.png)

Due to the flexibility of Windows 3.1 personalization control panel, I am able to re-color almost all standard Windows UI controls, including changing the color of highlights and shadows, allowing me to create a "flat", modern, dark theme. I used the UI colors of Blend for VS2017 and the highlight color of VS2017 to create a dark theme for Windows 3.1, which also applied to WinFile.

This screenshot is a live example (i.e. not photoshopped or concept art) showing WinFile sporting a dark theme. This can be used as reference to re-color the standard Windows components at the application level, which Windows 10 still supports.

Because WinFile uses native UI controls, it technically supports a dark theme color. And applications themselves can still apply their own color to these classic Windows UI components, including buttons, textboxes, etc (as long as they are not using a visual style, which WinFile isn't). So, it's not inconceivable that we would eventually be able to re-color the UI easily and without major modifications or re-skin to create the dark theme. The live screenshot is living proof that it works well with WinFile.

**Comment by T8z5h3** [comment]
looks great except for the issues note about the buttons.

**Comment by NazmusLabs** [comment]
Actually, most buttons in Windows 3.1 supports dark theme. The reason the drive button looks so bad here is that WinFile decided hard code the button highlights & shadows and used bitmaps for the icon. That's why I always prefer using as much native UI as possible. It keeps things future proof and allows better theming support down the line.

Here's how a button is supposed to look had it been created natively. The confirmation settings uses native controls, and, therefore, it correctly supports the dark theme. Notice that it looks flat, as I removed the highlight & shadow color to give Win31 a modern look.

![windows 3 1 file manager dark theme dosbox ss1801](https://user-images.githubusercontent.com/5572377/38569279-d51fc354-3cb8-11e8-9109-e05b5141aad9.png)

**Comment by NazmusLabs** [comment]
@Anixx In this case, you had to change your entire OS colors, which, Windows 10 and 8 doesn't support by default (and will ruin the look of other apps if you do it manually through reg edits). This thread is about an application specific color. Each app can recolor the standard Windows UI component for that app only, allowing you to always have a particular app to have a particular color, regardless of the color the OS is using. This only applies to the "classic theme" controls, which the WinFile is using.

P.S. notice how Windows 3.1's flat design allows the UI to be a lot more cleaner and less distracting without all the bevels, emboss, outlines, letting you focus on the content. Which is why I love modern UI paradigms.

**Comment by T8z5h3** [comment]
windows 10 does have a dark mode

**Comment by ZanderBrown** [comment]
@T8z5h3 for UWP apps and only if they haven't disabled support

This is the good old days of Win32 (well Win16) where you could change the colour and font of basically everything in the system and applications would respect it. Unfortunately the native controls where somewhat limited and complex to use (and remained like that until WinForms)  so applications and even parts of Windows itself moved to custom toolkits that were easier to work with (see Office or Media Player) and as such support was largely dropped

**Comment by T8z5h3** [comment]
i do remember that

**Comment by Amaroq-Clearwater** [comment]
I welcome these changes. There are a few other repositories which may interest you in relation to this pull request, but I forgot the specific URLs for them. I'll attach them later as soon as I can.

**Comment by haydennyyy** [comment]
Was that screenshot built on Win3.1? Because if not I really want that window style DX

**Comment by NazmusLabs** [comment]
@heydennyyy yup, it was built on vanilla Windows 3.1 using the built in control panel settings to change themes

**Comment by OrthodoxWindows** [comment]
> 

Another solution is to create a personalized dark visual style. Like that, all Win32 applications will support the dark mode.
The main problem is that the way Microsoft manages visual style is absurd; Instead of creating a separate "dark mod" visual style, Microsoft stupidly persists in creating a "dark mod" inside the "light" visual style, which most Win32 programs cannot take care of.

**Comment by pamochi** [comment]
@Anixx If you could explain how you got that theme to work with winfile, it would be greatly appreciated. Been looking around but haven't seen much and @NazmusLabs explanation for your post wasn't all to helpful

**Comment by Anixx** [comment]
@Anixx reggarding colors, it simply works, picking the colors from the system, you do not need to do anything.

**Comment by pamochi** [comment]
> @Anixx reggarding colors, it simply works, picking the colors from the system, you do not need to do anything.

I have my system colours set to grey and in the dark theme. Is there some directory your put the win file folder directly that could determine this. Right now I just have it in my program files folder, but maybe putting it in %system% could make the dark theme work

**Comment by Anixx** [comment]
It does not depend on directory, it depends on the system colors in your registry. For instance:
[colors.reg.txt](https://github.com/microsoft/winfile/files/13629052/colors.reg.txt)

**Comment by pamochi** [comment]
@Anixx I see what you mean. I assumed the default windows personalization would do this, but it seems not. It would be nice to be able to selectively control the controls for the apps as these edits mess with some of my other apps looks, but thanks for the help

**Comment by Anixx** [comment]
You can change all the colors in the system, including those not included in UWP settings app using WinPaletter.

**Comment by pamochi** [comment]
Just got the time to mess with WinPaletter, definitely a great tool for customization, so thank you for recommending it. There were two elements I still couldn't manage to change that I noticed you did, I attached a photo with those elements drawn over in red, if you had any idea. I took your reg file and imported it as well as tried winpaletter to alter these, though I could just be missing something
<img width="1440" alt="Screenshot 2023-12-10 162933" src="https://github.com/microsoft/winfile/assets/121124408/e37e8b94-43fb-44a8-9cdb-694a8b7a52de">
